src/js/view/utils/clickhandler.ts

```diff
@@ -1,17 +1,24 @@
 import { CLICK } from 'events/events';
 import UI from 'utils/ui';
 import Events from 'utils/backbone.events';
+import type Model from 'controller/model';
+import type { GenericObject } from 'types/generic.type';
 
-export default class ClickHandler {
-    constructor(model, element) {
-        Object.assign(this, Events);
+export default class ClickHandler extends Events {
+    domElement: HTMLElement | null;
+    model: Model | null;
+    ui: UI;
+    alternateClickHandler?: Function | null;
+    alternateDoubleClickHandler?: Function | null;
```

Comment:
How about instantiating these in the constructor to `null` so that the optional flag `?` can be dropped?